src/files/examples/fingerprint_reader_functions.py

```diff
@@ -0,0 +1,260 @@
+
+	    if ( f.verifyPassword() == False ):
+		raise ValueError('The given fingerprint sensor password is wrong!')
+
+	except Exception as e:
+	    print('The fingerprint sensor could not be initialized!')
+	    print('Exception message: ' + str(e))
+	    exit(1)
+
+def display_system_parameters():
+	""" Display system parameters as logger info level"""
+	tuple1 = f.getSystemParameters()
+	logging.info("System parameters : status register (%s), system id(%s), storage capacity(%s), security level(%s), sensor address(%s), packet length(%s), baudrate(%s)" % (tuple1[0], tuple1[1], tuple1[2], tuple1[3], tuple1[4], tuple1[5], tuple1[6]))
+
+def delete():
+	"""Deletes a template after asking for its position """
+	try:
+		positionNumber = input('Please enter the template position you want to delete: ')
+		positionNumber = int(positionNumber)
+
+		if ( f.deleteTemplate(positionNumber + 1) == True ):
+			print('Template deleted!')
+	except Exception as e:
+		print('Operation failed!')
+		print('Exception message: ' + str(e))
+		exit(1)
+
+def delete_all():
+	"""Deletes all templates after asking for confirmation"""
+	try:
+		answer = raw_input('Do you really want to delete all templates? y/n:')
+
+		if ( answer == "y" ):
+			f.clearDatabase()
+			print("Templates deleted! %s/%s" % (f.getTemplateCount(), f.getStorageCapacity()))
+	except Exception as e:
+		print('Operation failed!')
+		print('Exception message: ' + str(e))
+		exit(1)
+
+def download_image():
+	"""Downloads image of read fingerprint into temp directory"""
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		print('Downloading image (this take a while)...')
+
+		imageDestination =  tempfile.gettempdir() + '/fingerprint.bmp'
+		f.downloadImage(imageDestination)
+
+		print('The image was saved to "' + imageDestination + '".')
+
+	except Exception as e:
+		print('Operation failed!')
+		print('Exception message: ' + str(e))
+		exit(1)
+
+def search():
+	"""Searches for given fingerprint"""
+	## Tries to search the finger and calculate hash
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 1
+		logging.debug("Image read converting it")
+		f.convertImage(0x01)
+
+		## Searchs template
+		result = f.searchTemplate()
+
+		positionNumber = result[0]
+		accuracyScore = result[1]
+
+		if ( positionNumber == -1 ):
+			print('No match found!')
+			exit(0)
+		else:
+			print('Found template at position #' + str(positionNumber))
+			print('The accuracy score is: ' + str(accuracyScore))
+
+		## OPTIONAL stuff
+		##
+
+		## Loads the found template to charbuffer 1
+		f.loadTemplate(positionNumber, 0x01)
+
+		## Downloads the characteristics of template loaded in charbuffer 1
+		characteristics = str(f.downloadCharacteristics(0x01)).encode('utf-8')
```

Comment:
Maybe remove the `optional section` and add separate switch for downloading characteristics. What do you think @bastianraschke ?

Reply:
I think it has its place here, because it is a sample for people to understand what can be done with your lib...